src/spikeinterface/generation/splitting_tools.py

```diff
@@ -0,0 +1,139 @@
+import numpy as np
+from spikeinterface.core.numpyextractors import NumpySorting
+
+
+def split_sorting_by_times(
+    sorting_analyzer, splitting_probability=0.5, partial_split_prob=0.95, unit_ids=None, min_snr=None, seed=None
+):
+    """
+    Fonction used to split a sorting based on the times of the units. This
+    might be used for benchmarking meta merging step (see components)
+
+    Parameters
+    ----------
+    sorting_analyzer : A sortingAnalyzer object
+        The sortingAnalyzer object whose sorting should be splitted
+    splitting_probability : float, default 0.5
+        probability of being splitted, for any cell in the provided sorting
+    partial_split_prob : float, default 0.95
+        The percentage of spikes that will belong to pre/post splits
+    unit_ids : list of unit_ids, default None
+        The list of unit_ids to be splitted, if prespecified
+    min_snr : float, default=None
+        If specified, only cells with a snr higher than min_snr might be splitted
+    seed : int | None, default: None
+        The seed for random generator.
+
+    Returns
+    -------
+    new_sorting, splitted_pairs : The new splitted sorting, and the pairs that have been splitted
+    """
+
+    sa = sorting_analyzer
```

Comment:
you should use analyzer instead of sa for readability no ?

Reply:
Done